File: flutter/flutter-animated_barnacle/lib/User/UserModel.dart
class User {
  String id = "";
  String username;
  String email;
  String password;

  User({
    required id,
    required this.username,
    required this.email,
    required this.password, 
  });

  // Method to convert JSON data into a User object
  factory User.fromJson(Map<String, dynamic> json) {
    return User(
      id: json['_id'],
      username: json['username'],
      email: json['email'],
      password: json['password'],  // Avoid storing password in production
    );
  }

  // Method to convert a User object into JSON
  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'email': email,
      'password': password, // Avoid sending plain text passwords
    };
  }

}

File: react/react-animated-barnacle/src/Users/UserModel.js
class User {
    constructor(id, username, email, password) {
      this.id = id || ""; 
      this.username = username;
      this.email = email;
      this.password = password; 
    }
  
    // Method to convert JSON data into a User object
    static fromJson(json) {
      return new User(
        json._id,          
        json.username,
        json.email,
        json.password      
      );
    }
  
    // Method to convert a User object into JSON
    toJson() {
      return {
        username: this.username,
        email: this.email,
        password: this.password 
      };
    }
};

export default User;